pkg/controller/deorbit/events.go

```diff
@@ -16,6 +17,20 @@ type EventingDeorbiter struct {
 	Recorder  record.EventRecorder
 }
 
+func (d *EventingDeorbiter) DeleteSnapshots() (deleted []*unstructured.Unstructured, err error) {
+	deleted, err = d.Deorbiter.DeleteSnapshots()
+
+	for i, snapshot := range deleted {
+		if err == nil || i < (len(deleted)-1) {
```

Comment:
`i` is guaranteed to not exceed `len(deleted)` by the range expression. Also err needs to be only checked once outside of the loop.

Reply:
Can be simplified with an early return at `err != nil`.

Reply:
Hopefully this easier to grasp:
It first logs all successfully deleted objects and the failed on last if exists.